src/tracing/metric_macros.h

```diff
@@ -0,0 +1,283 @@
+        ::chip::Tracing::Internal::LogMetricEvent(_metric_event);                                                                  \
+    } while (false)
+
+////////////////////////
+// Metric logging macros
+////////////////////////
+
+/**
+ *  @def MATTER_LOG_METRIC
+ *
+ *  @brief
+ *    When tracing is enabled, this macro generates a metric event and logs it to the tracing backend.
+ *
+ *  Example usage:
+ *  @code
+ *      MATTER_LOG_METRIC(chip::Tracing::kMetricPASESession, err);
+ *  @endcode
+ *      The above example generates an instant metric event with key kMetricPASESession.
+ *      The metric also holds the 32 bit value corresponding to the ChipError object 'err'.
+ *
+ *  @param[in]  key The key representing the metric name/event.
+ *
+ *  @param[in]  value An optional value for the metric. This value corresponds to one of the values supported
+ *                    in MetricEvent::Value
+ */
+#define MATTER_LOG_METRIC(key, ...) __MATTER_LOG_METRIC(chip::Tracing::MetricEvent::Type::kInstantEvent, key, ##__VA_ARGS__)
+
+/**
+ * @def MATTER_LOG_METRIC_BEGIN
+ *
+ * @brief
+ * Generate a metric with the Begin Type
+ *
+ *  Example usage:
+ *  @code
+ *      MATTER_LOG_METRIC_BEGIN(chip::Tracing::kMetricPASESession);
+ *  @endcode
+ *      The above example generates a Begin metric event with key kMetricPASESession.
+ *
+ *  @param[in]  key The key representing the metric name/event.
+ *
+ *  @param[in]  value An optional value for the metric. This value corresponds to one of the values supported
+ *                    in MetricEvent::Value
+ */
+#define MATTER_LOG_METRIC_BEGIN(key, ...) __MATTER_LOG_METRIC(chip::Tracing::MetricEvent::Type::kBeginEvent, key, ##__VA_ARGS__)
+
+/**
+ * @def MATTER_LOG_METRIC_END
+ *
+ * @brief
+ * Generate a metric with the End Type
+ *
+ *  Example usage:
+ *  @code
+ *      MATTER_LOG_METRIC_END(chip::Tracing::kMetricPASESession);
+ *  @endcode
+ *      The above example generates an End metric event with key kMetricPASESession.
+ *
+ *  @param[in]  key The key representing the metric name/event.
+ *
+ *  @param[in]  value An optional value for the metric. This value corresponds to one of the values supported
+ *                    in MetricEvent::Value
+ */
+#define MATTER_LOG_METRIC_END(key, ...) __MATTER_LOG_METRIC(chip::Tracing::MetricEvent::Type::kEndEvent, key, ##__VA_ARGS__)
+
+/**
+ * @def MATTER_LOG_METRIC_SCOPE
+ *
+ * @brief
+ * Generate a scoped metric tracking Begin and End within a given scope.
+ *
+ *  Example usage:
+ *  @code
+ *      MATTER_LOG_METRIC_SCOPE(chip::Tracing::kMetricPASESession);
+ *  @endcode
+ *      The above example generates an Begin and the End metric using RAII.
+ *
+ *  @param[in]  key The key representing the metric name/event.
+ */
+#define MATTER_LOG_METRIC_SCOPE(key)                                                                                               \
```

Comment:
this does not need to be a metric: there is no metric value.

Reply:
Please see comments above on the difference